include/xtensor/xsort.hpp

```diff
@@ -0,0 +1,213 @@
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#ifndef XTENSOR_SORT_HPP
+#define XTENSOR_SORT_HPP
+
+#include <iterator>
+#include <algorithm>
+
+#include "xarray.hpp"
+#include "xeval.hpp"
+#include "xstridedview.hpp"
+#include "xslice.hpp"  // for xnone
+#include "xtensor.hpp"
+
+namespace xt
+{
+    template <class E>
+    auto sort(const xexpression<E>& e)
+    {
+        const auto& de = e.derived_cast();
+        return sort(de, de.dimension() - 1);
+    }
+
+    template <class E>
+    auto sort(const xexpression<E>& e, placeholders::xtuph /*t*/)
+    {
+        using value_type = typename E::value_type;
+        const auto de = e.derived_cast();
+        xtensor<value_type, 1> ev;
+        ev.reshape({de.size()});
+
+        std::copy(de.begin(), de.end(), ev.begin());
+        std::sort(ev.begin(), ev.end());
+
+        return ev;
+    }
+
+    template <class E, class F>
+    void call_over_last_axis(E& ev, F&& fct)
+    {
+        using value_type = typename E::value_type;
+
+        ptrdiff_t secondary_stride = (ptrdiff_t)  ev.strides()[ev.dimension() - 2];
+        ptrdiff_t offset = 0;
+
+        std::size_t n_iters = 1;
+        for (std::size_t i = 0; i < ev.dimension() - 1; ++i)
+        {
+            n_iters *= ev.shape()[i];
+        }
+
+        for (std::size_t i = 0; i < n_iters; ++i, offset += secondary_stride)
+        {
+            fct(ev.raw_data() + offset, ev.raw_data() + offset + secondary_stride);
+        }
+    }
+
+    /**
+     * Sort xexpression (optionally along axis)
+     * The sort is performed using the ``std::sort`` functions.
+     * A copy of the xexpression is created and returned.
+     *
+     * @param e xexpression to sort
+     * @param axis axis along which sort is performed
+     *
+     * @return sorted array (copy)
+     */
+    template <class E>
+    auto sort(const xexpression<E>& e, std::size_t axis)
+    {
+        using eval_type = typename E::temporary_type;
+        using value_type = typename E::value_type;
+
+        const auto& de = e.derived_cast();
+
+        auto axis_numbers = arange<std::size_t>(de.shape().size());
+        std::vector<std::size_t> permutation(axis_numbers.begin(), axis_numbers.end());
+        permutation.erase(permutation.begin() + (ptrdiff_t) axis);
+        permutation.push_back(axis);
+
+        std::vector<std::size_t> reverse_permutation;
+        for (auto el : axis_numbers)
+        {
+            auto it = std::find(permutation.begin(), permutation.end(), el);
+            reverse_permutation.push_back((std::size_t) std::distance(permutation.begin(), it));
+        }
+
+        eval_type ev = transpose(de, permutation);
```

Comment:
@serge-sans-paille I finally found a good use for the permutation parameter of transpose ;)